src/controllers/pagoPAController.ts

```diff
@@ -5,65 +5,99 @@
 import { PagoPAUser } from "../../generated/pagopa/PagoPAUser";
-
 import ProfileService from "../services/profileService";
-import { notFoundProfileToInternalServerError } from "../types/profile";
 import { withUserFromRequest } from "../types/user";
 
-export default class PagoPAController {
-  constructor(private readonly profileService: ProfileService) {}
+const VALIDATION_ERROR_TITLE = "Validation Error";
 
+export default class PagoPAController {
+  constructor(readonly profileService: ProfileService) {}
   /**
    * Returns the profile for the user identified by the provided fiscal
    * code.
+   * @deprecated by https://www.pivotaltracker.com/story/show/170259398
    */
   public readonly getUser = (
     req: express.Request
   ): Promise<
     // tslint:disable-next-line:max-union-size
     | IResponseErrorValidation
     | IResponseErrorInternal
-    | IResponseErrorTooManyRequests
     | IResponseSuccessJson<PagoPAUser>
   > =>
-    withUserFromRequest(req, async user => {
-      const response = notFoundProfileToInternalServerError(
-        await this.profileService.getProfile(user)
-      );
+    withUserFromRequest(req, async user =>
+      PagoPAUser.decode({
+        email: user.spid_email,
+        family_name: user.family_name,
+        mobile_phone: user.spid_mobile_phone,
+        name: user.name
+      }).fold<IResponseErrorValidation | IResponseSuccessJson<PagoPAUser>>(
+        _ =>
+          ResponseErrorValidation(VALIDATION_ERROR_TITLE, "Invalid User Data"),
+        pagopaUser => ResponseSuccessJson(pagopaUser)
+      )
+    );
 
+  /**
+   * Returns the profile for the user identified by the provided fiscal
+   * code. Notice email and Fiscal Code are provided.
+   */
+  public readonly getExtendedUser = (
+    req: express.Request
+  ): Promise<
+    // tslint:disable-next-line:max-union-size
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseErrorInternal
+    | IResponseErrorTooManyRequests
+    | IResponseSuccessJson<ExtendedPagoPAUser>
+  > =>
+    withUserFromRequest(req, async user => {
+      const response = await this.profileService.getProfile(user);
       if (response.kind !== "IResponseSuccessJson") {
         // if getProfile returns a failure, we just return it
         return response;
       }
-
-      // getProfile returns an InitializedProfile
       const profile = response.value;
 
-      // a custom email may have been set in the InitializedProfile, thus we
-      // have to check if the profile it's an InitializedProfile to be able to
-      // retrieve it
-      if (!profile.email || !profile.is_email_validated) {
-        return ResponseErrorValidation(
-          "Validation Error",
-          "Missing User Email"
-        );
-      }
+      // a validated custom email may have been set in the InitializedProfile,
+      // this email must be used only if is present and validated.
+      const maybeCustomEmailValidated =
```

Comment:
this is ok for now. generally speaking this kind of constraints are better expressed with types, ie:

```ts
const ProfileWithValidCustomEmail =  t.interface({ ..., 
 is_email_validated: t.literal(true), email: EmailFromString 
})
const ProfileWithSpidEmail =  t.interface({ ..., 
  spid_email: EmailFromString 
})
```

this avoids to pass around a profile object which has an inconsistent states (ie. `{ email: undefined, is_email_validated: true }`)